internal/transport/mqtt.go

```diff
@@ -107,6 +108,13 @@ func (t *MQTT) Connect() error {
 	return nil
 }
 
+// Reload reloads TLS config and started a new client with the new tls.Config
+func (t *MQTT) Reload(tlsConfig *tls.Config) error {
+	t.opts.SetTLSConfig(tlsConfig.Clone())
+	t.client = mqtt.NewClient(t.opts)
```

Comment:
good catch! I need to check!

Reply:
Finally, I had time to check this, what I'm doing is disconnected the old client and connecting a new one.  Overall is working perfect and all is subscribed again because all the info on startup is stored on t.opts.